PetAdoptions/petsite/petsite/Controllers/PaymentController.cs

```diff
@@ -1,168 +1,104 @@
-            
-            */
-           // Console.WriteLine($"STEPLOG -ARN - {SystemsManagerConfigurationProviderWithReloadExtensions.GetConfiguration(_configuration,"petadoptionsstepfnarn")}");
-            //Console.WriteLine($"STEPLOG - SERIALIZE - {JsonSerializer.Serialize(new SearchParams() {petid = petId, pettype = petType})}");
-            AWSSDKHandler.RegisterXRay<IAmazonStepFunctions>();
-            return await new AmazonStepFunctionsClient().StartExecutionAsync(new StartExecutionRequest()
-            {
-                Input = JsonSerializer.Serialize(new SearchParams() {petid = petId, pettype = petType}),
-                Name = $"{petType}-{petId}-{Guid.NewGuid()}",
-                StateMachineArn = SystemsManagerConfigurationProviderWithReloadExtensions.GetConfiguration(_configuration,"petadoptionsstepfnarn")
-            });
-        }
-
-        private async Task<PublishResponse> SendNotification(string petId)
-        {
-            AWSSDKHandler.RegisterXRay<IAmazonService>();
-
-            var snsClient = new AmazonSimpleNotificationServiceClient();
-            return await snsClient.PublishAsync(topicArn: _configuration["snsarn"],
-                message: $"PetId {petId} was adopted on {DateTime.Now}");
-        }
-    }
-}
\ No newline at end of file
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Amazon.SQS;
+using Microsoft.Extensions.Configuration;
+using Prometheus;
+
+namespace PetSite.Controllers
+{
+    public class PaymentController : BaseController
+    {
+        private static string _txStatus = String.Empty;
+        
+        private readonly ILogger<PaymentController> _logger;
+
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IConfiguration _configuration;
+
+        //Prometheus metric to count the number of Pets adopted
+        private static readonly Counter PetAdoptionCount =
+            Metrics.CreateCounter("petsite_petadoptions_total", "Count the number of Pets adopted");
+
+        public PaymentController(ILogger<PaymentController> logger, IConfiguration configuration, IHttpClientFactory httpClientFactory)
+        {
+            _configuration = configuration;
+            _httpClientFactory = httpClientFactory;
+            _logger = logger;
+        }
+
+        // GET: Payment
+        [HttpGet]
+        public ActionResult Index()
+        {
+            if (EnsureUserId()) return new EmptyResult();
+            
+            // Transfer TempData to ViewData for the view
+            if (TempData["txStatus"] != null)
+            {
+                ViewData["txStatus"] = TempData["txStatus"];
+                ViewData["error"] = TempData["error"];
+            }
+            
+            return View();
+        }
+
+        // POST: Payment/MakePayment
+        [HttpPost]
+        // [ValidateAntiForgeryToken]
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Cross-Site Request Forgery (CSRF) protection is not implemented or is disabled. 
This vulnerability allows attackers to trick users into performing unwanted actions 
on your site when they&#39;re authenticated. Implement proper CSRF protection using 
`ValidateAntiForgeryToken` or `AutoValidateAntiforgeryToken`.
Learn more: https://cwe.mitre.org/data/definitions/352.html